docs/backend/training-repo.md

```diff
@@ -0,0 +1,84 @@
+# New Repository Proposal: Training
+
+## Summary
+
+This document proposes a new repository under the `instructlab` GitHub organization:
+
+- `instructlab/training`
+
+## Background
+
+Today, we have training implemented for different use-cases
+existing in different repos. LoRA/QLoRA training currently lives
+in the [`instructlab`](https://github.com/instructlab/instructlab)
+CLI repo and contains logic for both Linux and MacOS training.
+Each of these then brings additional logic for platform-specific accelerators.
+For example NVIDIA CUDA and Intel Habana on Linux, and MLX and MPS on MacOS.
+
+On the other hand, we have logic that exists today for full fine-tuning
+in the [instructlab/training](https://github.com/instructlab/training) repo. Today, this implementation is CUDA-specific
+as it relies on NCCL for handling communication across various
+NVIDIA GPUs.
+We want to bring support for alternative accelerators such as Intel Gaudi to this repo as well, without rewriting much of the logic.
+
+
+
+We need to have a library which provides a common interface
+for training, allowing all of the hard logic to be concentrated
+in a single point.
+
+The [instructlab/training](https://github.com/instructlab/training) repo should then
+become the home for the overall training library.
```

Comment:
just to be clear -- this proposal is to expand the scope of the current `training` repo to be the `training` library that the CLI will consume?

Reply:
Yeah, I should have made that clear. We're just reworking the existing training repo to now contain the new interfaces.

Reply:
Updated to address this.

Reply:
thanks